pydantic/_internal/_generate_schema.py

```diff
@@ -1176,36 +1176,51 @@ def apply_model_validators(
 
 
 def apply_annotations(
-    schema: core_schema.CoreSchema, annotations: typing.Iterable[Any], definitions: dict[str, core_schema.CoreSchema]
+    schema: core_schema.CoreSchema,
+    annotations: typing.Iterable[Any],
+    definitions: dict[str, core_schema.CoreSchema],
+    source_type: Any,
 ) -> core_schema.CoreSchema:
     """
     Apply arguments from `Annotated` or from `FieldInfo` to a schema.
     """
     handler = CoreMetadataHandler(schema)
-    for metadata in annotations:
-        schema = apply_single_annotation(schema, metadata, definitions)
+
+    def expand_grouped_metadata(annotation: Any) -> Iterator[Any]:
+        if isinstance(annotation, GroupedMetadata):
```

Comment:
Is `FieldInfo` a `GroupedMetadata`? Should check and add a test if this is a bug.